src/Components/Nav/MobileTopNav.jsx

```diff
@@ -0,0 +1,195 @@
+import React, { useState, useEffect, useCallback } from "react";
+import { useTranslation } from "react-i18next";
+import { useSelector, useDispatch } from "react-redux";
+import { Link, useLocation } from "react-router-dom";
+
+import AccountCircle from "@material-ui/icons/AccountCircle";
+import AddIcon from "@material-ui/icons/Add";
+import BarChartIcon from "@material-ui/icons/BarChart";
+import ExitToAppIcon from "@material-ui/icons/ExitToApp";
+import InfoIcon from "@material-ui/icons/Info";
+import LibraryBooksIcon from "@material-ui/icons/LibraryBooks";
+import LocalLibraryIcon from "@material-ui/icons/LocalLibrary";
+import LockIcon from "@material-ui/icons/Lock";
+import PolicyIcon from "@material-ui/icons/Policy";
+import SecurityIcon from "@material-ui/icons/Security";
+import SettingsIcon from "@material-ui/icons/Settings";
+import StyleIcon from "@material-ui/icons/Style";
+import VerifiedUserIcon from "@material-ui/icons/VerifiedUser";
+
+import useLinkCreator from "../../hooks/useLinkCreator.js";
+import { signOut } from "../../redux/Actions/login.js";
+
+import { ReactComponent as HamburgerMenuIcon } from "../../images/icons/hamburger.svg";
+import VocascanLogo from "../../images/logo/color-round.svg";
+
+import "./MobileTopNav.scss";
+
+import { useScrollBlock } from "../../hooks/useScrollBlock";
+import { pages } from "../../utils/constants";
+import NavButton from "./NavButton";
+
+const MobileTopNav = () => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const [blockScroll, allowScroll] = useScrollBlock();
+
+  const location = useLocation();
+
+  const [showNav, setShowNav] = useState(true);
+
+  const [hamburgerOpen, setHamburgerOpen] = useState(false);
+
+  const isAdmin = useSelector((state) => state.login.user.isAdmin);
+
+  const { isValid: isLegalNoticeValid, url: legalNoticeUrl } = useLinkCreator({
+    path: pages.legalNotice,
+    electronFix: true,
+  });
+  const { isValid: isTermsAndConditionsValid, url: termsAndConditionsUrl } =
+    useLinkCreator({ path: pages.termsAndConditions });
+  const { isValid: isPrivacyPolicyValid, url: privacyPolicyUrl } =
+    useLinkCreator({ path: pages.privacyPolicy });
+
+  const triggerHamburgerMenu = useCallback(() => {
+    setHamburgerOpen(!hamburgerOpen);
+  }, [hamburgerOpen]);
+
+  const closeHamburgerMenu = useCallback(() => {
+    setHamburgerOpen(false);
+  }, []);
+
+  const listenScrollEvent = useCallback(() => {
+    if (window.scrollY > 10) {
+      setShowNav(false);
+    } else {
+      setShowNav(true);
+    }
+  }, []);
+
+  const handleLogout = useCallback(() => {
+    dispatch(signOut());
+  }, [dispatch]);
+
+  useEffect(() => {
+    window.addEventListener("scroll", listenScrollEvent);
+
+    return () => window.removeEventListener("scroll", listenScrollEvent);
+  }, [listenScrollEvent]);
+
+  useEffect(() => {
+    hamburgerOpen ? blockScroll() : allowScroll();
```

Comment:
why do we need to block the scroll here?

Reply:
This block was intended to block the content, that is lying below the MobileTopNav, as you can still scroll it, while the nav is open. In the end you could just remove this function, but I just kopied it, because I have built it in the homepage too. Should I remove it

Reply:
No, we can leave it if it has some advantages.